src/opm/parser/eclipse/EclipseState/Schedule/VFPProdTable.cpp

```diff
@@ -388,21 +376,21 @@ void VFPProdTable::check(const DeckKeyword& keyword, const double table_scaling_
     //Check that bhp(thp) is a monotonic increasing function.
     //If this is not the case, we might not be able to determine
     //the thp from the bhp easily
-    std::string points = "";
-    for (size_type w=0; w<m_data.shape()[1]; ++w) {
-        for (size_type g=0; g<m_data.shape()[2]; ++g) {
-            for (size_type a=0; a<m_data.shape()[3]; ++a) {
-                for (size_type f=0; f<m_data.shape()[4]; ++f) {
-                    double bhp_last = m_data[0][w][g][a][f];
-                    for (size_type t=0; t<m_data.shape()[0]; ++t) {
-                        if (m_data[t][w][g][a][f] < bhp_last) {
+    std::string points;
+    for (size_type w = 0; w < nw; ++w) {
+        for (size_type g = 0; g < ng; ++g) {
+            for (size_type a = 0; a < na; ++a) {
+                for (size_type f = 0; f < nf; ++f) {
+                    double bhp_last = (*this)(0,w,g,a,f);
+                    for (size_type t = 0; t < nt; ++t) {
+                        if ((*this)(t,w,g,a,f) < bhp_last) {
```

Comment:
I guess the `std::isnan()` check could be moved here - to remove one of the five level for loops.

Reply:
done.